assets/src/api.ts

```diff
@@ -441,6 +441,22 @@ export const putRouteTabs = (routeTabs: RouteTab[]): Promise<Response> =>
     body: JSON.stringify({ route_tabs: routeTabs }),
   })
 
+export const putDetourUpdate = (snapshot: any): Promise<string | null> =>
+  checkedApiCall({
+    url: `/api/detours/update_snapshot`,
+    parser: nullableParser((id: string) => id),
+    dataStruct: any(),
+    defaultResult: null,
+    fetchArgs: {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+        "x-csrf-token": getCsrfToken(),
+      },
+      body: JSON.stringify({ snapshot: snapshot }),
+    },
+  })
```

Comment:
suggestion: Can we use `apiCallResult` instead? This would make the error state a lot more explicit.

I can post a code example later if needed.

Reply:
```suggestion
  apiCallResult(
    `/api/detours/update_snapshot`,
    object({ data: string() }),
    any(),
    {
      method: "PUT",
      headers: {
        "Content-Type": "application/json",
        "x-csrf-token": getCsrfToken(),
      },
      body: JSON.stringify({ snapshot: snapshot }),
    }
  )
```

Reply:
Gotcha. (I was following in the predominant pattern of the rest of the api functions.)

So, I see `apiCallResult` is a recent add, but I'm not sure what new-and-different thing it does as compared to the other call methods? I'm thinking `apiCall`, `checkedApiCall`, and `apiCallResult` also need comments, especially if we're trying to shift towards a newer function over time

Reply:
Documentation! https://github.com/mbta/skate/pull/2747